src/cets.erl

```diff
@@ -601,12 +613,17 @@ handle_get_info(
     #{
         table => Tab,
         nodes => lists:usort(pids_to_nodes([self() | Servers])),
-        size => ets:info(Tab, size),
-        memory => ets:info(Tab, memory),
+        size => returns_non_neg_integer(ets:info(Tab, size)),
+        memory => returns_non_neg_integer(ets:info(Tab, memory)),
         ack_pid => AckPid,
         opts => Opts
     }.
 
+%% We are sure that the ETS table exists when running handle_get_info
+%% But gradualizer does not know that, so let us tell it with the spec
+-spec returns_non_neg_integer(any()) -> non_neg_integer().
+returns_non_neg_integer(X) -> X.
+
```

Comment:
If you don't want/need the guard, i.e. the dynamic check (which I think is better to have in general, unless you're 100% the result cannot be anything else), you can also do:

```erlang
-include_lib("gradualizer/include/gradualizer.hrl").

...
size => ?assert_type(ets:info(Tab, size), integer())
...
```

Reply:
oh, actually runtime check
```
-spec assert_non_neg_integer(any()) -> non_neg_integer().
assert_non_neg_integer(X) when is_integer(X), X >= 0 -> X.
```

makes dialyzer unhappy

```
src/cets.erl
Line 620 Column 2: Type specification cets:assert_non_neg_integer(any()) -> non_neg_integer() is a supertype of the success typing: cets:assert_non_neg_integer(non_neg_integer()) -> non_neg_integer()
```
:)

Reply:
also, don't wanna add gradualizer as a build dep just for the header, to make compilation faster :D

Reply:
Yeah, we might add a `gradualizer_support` package at some point to address the compilation issue.